### PR TITLE
Initial attempt

### PR DESCRIPTION
**Pull request by Babakfifoo**
Hi!

I started the assignment, but I stuck in understanding the question itself 😅.

Currently, I have created the functions for detrending data and calculated pairwise correlation. But I do not understand the model RDM and then calculate RSA. 

I would appreciate it if you can help me with that 😃.

### CODE REVIEW

**Comment by wmvanvliet** [comment]
Want to hop into the course Zoom room (same we used during the lecture) and discuss live?

**Comment by wmvanvliet** [comment]
also, make sure to study our implementation document (including the tips section) carefully.

File: Assignment_1.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+bold = nib.load(data_dir + bold_dta_path)
+mask = nib.load(data_dir + mask_path)
+labeled_df = pd.read_csv(data_dir + label_path, sep=' ')
+
+# Masking the data
+
+voxels_in_roi = bold.get_fdata()[mask.get_fdata() == 1, :]
+
+# producing a dataframe with columns for each voxel.
+
```
Comment (wmvanvliet) [inline Assignment_1.py:131]: I recommend keeping the data as a 4D numpy array and not try to shoehorn this into a 2D dataframe.

File: Assignment_1.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+from scipy import signal
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import warnings
+# %%
+
+
+def detrend_polynomial(dta):
```
Comment (wmvanvliet) [inline Assignment_1.py:20]: Only a simple linear detrending is asked for in the assignment. It's a one liner with [`scipy.signal.detrend`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.detrend.html) if you keep the data as a numpy array. Once you start packing it as a pandas `DataFrame` you are making things difficult for yourself.

Comment (Babakfifoo) [reply Assignment_1.py:20]: Good to know. I will try the linear detrend. That makes my work much easier 😄😄.

**Comment by wmvanvliet** [comment]
About the model RDM: it is to be constructed from the `labels` column of the `labels.txt` file. For each MRI-image in the recording, there is a label describing which kind of photograph was being shown to the person in the scanner. The model  RDM is a 2D array (=a matrix) of shape n_brain_images x n_brain_images. So it contains all possible combinations of two brain images. When two brain images have the same label, the model RDM should contain a 0 in the corresponding cell. If not, it should contain a 1. If you first order the brain images and labels by the `labels` column, this matrix looks like this:

https://github.com/susamerz/CDWAssignment/blob/main/images/model_rdm.png

In short: an RDM is a matrix that contains for each possible combination of two brain images, a number representing how different these brain images are. The model RDM sketches an idealized situation where brain images during which the same type of photograph was shown, are identifical (=a difference value of 0) and if different types of photographs were shown, are completely different (=a difference value of 1). RSA analysis means comparing this idealized RDM to the actual RDM computed on the brain images themselves.
